deepinv/tests/test_loss.py

```diff
@@ -365,15 +367,23 @@ def test_losses(
         plot_images=(loss_name == LOSSES[0]),  # save time
         verbose=False,
         log_train_batch=(loss_name == "sup_log_train_batch"),
-        disable_train_metrics=(loss_name == "reducedresolution"),
     )
 
     # test the untrained model
-    initial_test = trainer.test(test_dataloader=test_dataloader)
+    initial_test = trainer.test(
+        test_dataloader=test_dataloader, metrics=deepinv.metric.PSNR()
+    )
+
+    # in self-supervised cases, remove supervised metrics and compute self-sup losses on eval dataset
+    trainer.metrics = (
```

Comment:
It makes sense, but having to tell the user to remove the metric seems a bit hacky. Not sure what the solution is here. Note that [Metrics already return NaN](https://github.com/deepinv/deepinv/blob/main/deepinv/loss/metric/metric.py#L184) when the data is None, so the trainer could automatically use the self-sup loss if the metric returns NaN. This doesn't account for the case, on the other hand, when there is GT in the dataset but you simply don't want to cheat and look at it for self-sup eval

Reply:
Resolving this issue because the conversation continues in https://github.com/deepinv/deepinv/pull/777#discussion_r2404190832